src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/SafeMsQuicConfigurationHandle.cs

```diff
@@ -147,15 +147,18 @@ private static unsafe SafeMsQuicConfigurationHandle Create(QuicOptions options,
             settings.IsSet.PeerBidiStreamCount = 1;
             settings.PeerBidiStreamCount = (ushort)options.MaxBidirectionalStreams;
 
-            settings.IsSet.IdleTimeoutMs = 1;
-            if (options.IdleTimeout != Timeout.InfiniteTimeSpan)
+            if (options.IdleTimeout != TimeSpan.Zero)
             {
-                if (options.IdleTimeout <= TimeSpan.Zero) throw new Exception("IdleTimeout must not be negative.");
-                settings.IdleTimeoutMs = (ulong)options.IdleTimeout.TotalMilliseconds;
-            }
-            else
-            {
-                settings.IdleTimeoutMs = 0;
+                settings.IsSet.IdleTimeoutMs = 1;
+                if (options.IdleTimeout != Timeout.InfiniteTimeSpan)
+                {
+                    if (options.IdleTimeout <= TimeSpan.Zero) throw new Exception("IdleTimeout must not be negative.");
```

Comment:
This should throw something derived from Exception, and the message should be in the resx.

Reply:
I'll solve this in #71783 since that touches this part more extensively.